tests/benchmark/test_bloatnet.py

```diff
@@ -0,0 +1,618 @@
+    Account,
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    Bytecode,
+    Hash,
+    Storage,
+    Transaction,
+    While,
+    compute_create2_address,
+    keccak256,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+REFERENCE_SPEC_GIT_PATH = "DUMMY/eip-DUMMY.md"
+REFERENCE_SPEC_VERSION = "0.1"
+
+# Constants for CREATE2 address calculation
+CREATE2_MEMORY_OFFSET = 0x1000  # Offset for CREATE2 calculation to avoid collision
+
+
+def generate_create2_address_calculation(
+    factory_address,
+    init_code_hash: bytes,
+) -> Bytecode:
+    """Generate EVM bytecode to calculate a CREATE2 address with salt on stack."""
+    code = Bytecode()
+
+    # Memory layout at CREATE2_MEMORY_OFFSET:
+    # [0xFF][factory_address][salt][init_code_hash]
+    base = CREATE2_MEMORY_OFFSET
+
+    # Store 0xFF prefix at memory[base]
+    code += Op.PUSH1(0xFF) + Op.PUSH2(base) + Op.MSTORE8
+
+    # Store factory address at memory[base+1:base+21]
+    # Handle both string and Address types
+    if isinstance(factory_address, str):
+        addr_hex = factory_address[2:] if factory_address.startswith("0x") else factory_address
+        factory_bytes = bytes.fromhex(addr_hex)
+    else:
+        # Assume it's an Address object with bytes representation
+        factory_bytes = bytes(factory_address)
+    code += Op.PUSH20(int.from_bytes(factory_bytes, "big")) + Op.PUSH2(base + 1) + Op.MSTORE
+
+    # Store salt at memory[base+21:base+53] (base+0x15)
+    # Assumes salt is already on stack
+    code += Op.PUSH2(base + 0x15) + Op.MSTORE
+
+    # Store init code hash at memory[base+53:base+85] (base+0x35)
+    code += Op.PUSH32(int.from_bytes(init_code_hash, "big")) + Op.PUSH2(base + 0x35) + Op.MSTORE
+
+    # Calculate keccak256 of 85 bytes starting at memory[base]
+    code += Op.PUSH1(0x55) + Op.PUSH2(base) + Op.SHA3
+
+    # The address is the last 20 bytes (already on stack)
+    return code
+
+
+@pytest.mark.valid_from("Prague")
+@pytest.mark.parametrize("final_storage_value", [0x02 << 248, 0x02])
+def test_bloatnet(
+    blockchain_test: BlockchainTestFiller,
+    pre: Alloc,
+    fork: Fork,
+    final_storage_value: int,
+    gas_benchmark_value: int,
+):
+    """
+    A test that calls a contract with many SSTOREs.
+
+    The first block will have many SSTORES that go from 0 -> 1
+    and the 2nd block will have many SSTORES that go from 1 -> 2
+    """
+    # Get gas costs for the current fork
+    gas_costs = fork.gas_costs()
+
+    # this is only used for computing the intinsic gas
+    data = final_storage_value.to_bytes(32, "big").rstrip(b"\x00")
+
+    storage = Storage()
+
+    # Initial gas for PUSH0 + CALLDATALOAD + POP (at the end)
+    totalgas = gas_costs.G_BASE * 2 + gas_costs.G_VERY_LOW
+    totalgas = totalgas + fork.transaction_intrinsic_cost_calculator()(calldata=data)
+    gas_increment = gas_costs.G_VERY_LOW * 2 + gas_costs.G_STORAGE_SET + gas_costs.G_COLD_SLOAD
+    sstore_code = Op.PUSH0 + Op.CALLDATALOAD
```

Comment:
PUSH0 is 2 gas and CALLDATALOAD is 3 gas, so I believe `gas_increment` is not entirely correct here (but the size of `G_STORAGE_SET` and `G_COLD_SLOAD` are so large that this one-off likely does not yield a bug)